Tests/Unit/Model/RatepayInvoicePaymentMethodTest.php

```diff
@@ -342,4 +344,97 @@ public function onBeforeOrderCreationFailedProvider()
+     */
+    public function testGetPostProcessingTransaction($aOrderItems)
+    {
+        $oParentTransaction = $this->getResponseXMLTransaction();
+
+        $oResult = $this->_oPaymentMethod->getPostProcessingTransaction(Transaction::ACTION_CREDIT, $oParentTransaction, $aOrderItems);
+
+        $this->assertInstanceOf(RatepayInvoiceTransaction::class, $oResult);
+    }
+
+    public function getPostProcessingTransactionProvider()
+    {
+        return [
+            'articles left' => [
+                ['Article Number' => 1],
+            ],
+            'no articles left' => [
+                ['Article Number' => 0],
+            ],
+            'other article left' => [
+                ['Other Article' => 1],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider getPostProcessingTransactionQuantityProvider
+     */
+    public function testGetPostProcessingTransactionQuantity($aOrderItems, $iExpected)
+    {
+        $oParentTransaction = $this->getResponseXMLTransaction();
+
+        $oResult = $this->_oPaymentMethod->getPostProcessingTransaction(Transaction::ACTION_CREDIT, $oParentTransaction, $aOrderItems);
+        $oResult->setOperation(Transaction::ACTION_CREDIT);
+
+        $this->assertEquals($iExpected, $oResult->mappedProperties()['order-items']['order-item'][0]['quantity']);
+    }
+
+    public function getPostProcessingTransactionQuantityProvider()
+    {
+        return [
+            'articles left' => [
+                ['Article Number' => 1],
+                1
+            ],
+            'no articles left' => [
+                ['Article Number' => 0],
+                null
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider getPostProcessingTransactionArticleProvider
+     */
+    public function testGetPostProcessingTransactionArticle($aOrderItems, $sExpected)
+    {
+        $oParentTransaction = $this->getResponseXMLTransaction();
+
+        $oResult = $this->_oPaymentMethod->getPostProcessingTransaction(Transaction::ACTION_CREDIT, $oParentTransaction, $aOrderItems);
+        $oResult->setOperation(Transaction::ACTION_CREDIT);
+
+        $this->assertEquals($sExpected, $oResult->mappedProperties()['order-items']['order-item'][0]['article-number']);
+    }
+
+    public function getPostProcessingTransactionArticleProvider()
+    {
+        return [
+            'articles left' => [
+                ['Article Number' => 1],
+                'Article Number'
+            ],
+            'no articles left' => [
+                ['Article Number' => 0],
+                null
+            ],
+        ];
+    }
+
+    private function getResponseXMLTransaction()
+    {
+        $oTransaction = $this->getMockBuilder(RatepayInvoiceTransaction::class)
+            ->setMethods(['getResponseXML'])
+            ->getMock();
+        $oTransaction
+            ->method('getResponseXML')
+            ->willReturn(file_get_contents(__DIR__ . '/../../resources/success_response.xml'));
```

Comment:
I saw this file being loaded 3x in the tests, is it posible to make a method in additional.inc.php or somwhere and then where its used just calling somthing like getSuccessXmlResponse() ? So if we make changes in the future ist one place to adopt it.